docs/examples/builds/multi-root-editor.md

```diff
@@ -0,0 +1,187 @@
+---
+category: examples-builds
+order: 60
+toc: false
+classes: main__content--no-toc
+---
+
+# Multi-root editor
+
+Multi-root editor is an editor type that features multiple, separate editable areas.
+
+The main difference between a multi-root editor and using multiple separate editors (like in the {@link examples/builds/inline-editor inline editor demo}) is the fact that in a multi-root editor all editable areas belong to the same editor instance and produce one document, share the same configuration, toolbar and the undo stack.
+
+{@snippet examples/multi-root-editor}
+
+## Editor example configuration
+
+Check out the {@link installation/getting-started/predefined-builds#installation-example Quick start} guide to learn more about implementing this kind of editor. You will find implementation steps there. You can see this example editor’s code below.
+
+<details>
+<summary>View editor configuration script</summary>
+
+```js
+import MultiRootEditor from '@ckeditor/ckeditor5-build-multi-root/src/ckeditor';
+
+MultiRootEditor
+	.create(
+		// Define roots / editable areas:
+		{
+			header: document.querySelector( '#header' ),
+			content: document.querySelector( '#content' ),
+			leftSide: document.querySelector( '#left-side' ),
+			rightSide: document.querySelector( '#right-side' )
+		},
+		// Editor configration:
+		{
+			cloudServices: {
+				// All predefined builds include the Easy Image feature.
+				// Provide correct configuration values to use it.
+				tokenUrl: 'https://example.com/cs-token-endpoint',
+				uploadUrl: 'https://your-organization-id.cke-cs.com/easyimage/upload/'
+				// Read more about Easy Image - https://ckeditor.com/docs/ckeditor5/latest/features/images/image-upload/easy-image.html.
+				// For other image upload methods see the guide - https://ckeditor.com/docs/ckeditor5/latest/features/images/image-upload/image-upload.html.
+			}
+		}
+	)
+	.then( editor => {
+		window.editor = editor;
+
+		// Append toolbar to a proper container.
+		const toolbarContainer = document.querySelector( '#toolbar' );
+		toolbarContainer.appendChild( editor.ui.view.toolbar.element );
+
+		// Make toolbar sticky when the editor is focused.
+		editor.ui.focusTracker.on( 'change:isFocused', () => {
+			if ( editor.ui.focusTracker.isFocused ) {
+				toolbarContainer.classList.add( 'sticky' );
+			} else {
+				toolbarContainer.classList.remove( 'sticky' );
+			}
+		} );
```

Comment:
I didn't make the toolbar sticky the way it is  done in the classic editor. It is possible and is not that difficult but it would lead to another decisions to be made. E.g. make it always sticky and by default, like in classic, or allow controling it? What if it doesn't make sense in some integrations?

Since `position: sticky` is widely supported, I decided to go this route.

If we enable sticky toolbar in multi-root editor, it should be also enabled for decopled editor, I think, to be consistent.

If it is done like in `ClassicEditor`, it seems it is possible to disable the stickiness through `StickyPanelView#isActive`. So, we can have it configurable. One thing I didn't like, though, is that you have to append `StickyPanelView` element, instead of `ToolbarView` element to the DOM. But maybe you would need toolbar instance for something. So, we may end up with `editor.ui.toolbarPanel` and `editor.ui.toolbar`. Additional confusion. Maybe `editor.ui.attachToolbarToDom( domContainer, isSticky )` would be a good idea.

cc @oleq.

Reply:
The reason why we went with our custom sticky panel implementation was to prevent the toolbar from staying above the editing root if you scroll past it like this:

```

 ┌───────────────────────────────────┐
 │                                   │
 │                                   │
 │     Editing root                  │
 │                                   │
 │                                   │
 └───────────────────────────────────┘


 ┌───────────────────────────────────┐
 │      Toolbar                      │
 └───────────────────────────────────┘

```

but keep it at a line-height distance from the bottom instead:

```

 ┌───────────────────────────────────┐
 │                                   │
 │                                   │
 │     Editing root                  │
 ├───────────────────────────────────┤
 │      Toolbar                      │
 ├───────────────────────────────────┤
 └───────────────────────────────────┘
```

If it wasn't for that, we'd go with pure CSS approach too. So I think you chose the right direction, at least for this demo. There are unlimited possibilities when it comes to the geometry of the multi-root editor, so hardcoding anything more would be an overkill IMO.